docs/resources/workspace_alert.md

```diff
@@ -0,0 +1,257 @@
+  display_name = "Sample Dashboard"
+  timeframe    = "last12hours"
+}
+
+# Extract ids of tiles
+locals {
+  dashboard_content = jsondecode(squaredup_dashboard.sample_dashboard.dashboard_content)
+
+  lambda_errors_tile    = [for content in local.dashboard_content.contents : content.i if content.config.title == "Lambda Errors"]
+  lambda_errors_tile_id = length(local.lambda_errors_tile) > 0 ? local.lambda_errors_tile[0] : null
+}
+
+data "squaredup_alerting_channel_types" "example" {
+  display_name = "Slack API"
+}
+
+resource "squaredup_alerting_channel" "slack_api_alert" {
+  display_name    = "Slack Alert - Team DevOps"
+  channel_type_id = data.squaredup_alerting_channel_types.example.alerting_channel_types[0].channel_id
+  config = jsonencode({
+    channel = "devops"
+    token   = "some-token"
+  })
+  enabled = true
+}
+
+// NOTE: Only one workspace alert resource can be created per workspace
+resource "squaredup_workspace_alert" "example" {
+  workspace_id = squaredup_workspace.application_workspace.id
+  alerting_rules = [
+    {
+      channel   = squaredup_alerting_channel.slack_api_alert.id
+      notify_on = "workspace_state"
+      // "workspace_state" does not support "preview_image"
+    },
+    {
+      channel       = squaredup_alerting_channel.slack_api_alert.id
+      preview_image = true
+      notify_on     = "all_monitors"
+    },
+    {
+      channel       = squaredup_alerting_channel.slack_api_alert.id
+      preview_image = false
+      notify_on     = "selected_monitors"
+      selected_monitors = [
+        {
+          dashboard_id = squaredup_dashboard.sample_dashboard.id
+          tiles_id     = [local.lambda_errors_tile_id]
+        }
+      ]
+    }
+  ]
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `alerting_rules` (Attributes List) The alerting rules to create (see [below for nested schema](#nestedatt--alerting_rules))
+- `workspace_id` (String) The ID of the workspace to create the alert in
+
+### Read-Only
+
+- `id` (String) The ID of the workspace
+
+<a id="nestedatt--alerting_rules"></a>
+### Nested Schema for `alerting_rules`
+
+Required:
+
+- `channel` (String) The ID of the channel to send the alert to
+- `notify_on` (String) Condition to trigger the alert. Must be one of: 'workspace_state', 'all_monitors', or 'selected_monitors'
+
+Optional:
+
+- `preview_image` (Boolean) Whether to include a preview image in the alert
+- `selected_monitors` (Attributes List) The monitors to trigger the alert on. Required if notify_on is 'selected_monitors' (see [below for nested schema](#nestedatt--alerting_rules--selected_monitors))
+
+<a id="nestedatt--alerting_rules--selected_monitors"></a>
+### Nested Schema for `alerting_rules.selected_monitors`
+
+Required:
+
+- `dashboard_id` (String) The ID of the dashboard where the monitor is configured
+- `tiles_id` (List of String) The ID of the tiles to trigger the alert on
```

Comment:
Shouldn't this be a list of objects where each object contains a dashboard id and a list of tile ids?  i.e. the selected monitors (tiles) can be on different dashboards within the workspace.
See https://github.com/squaredup/SquaredUpAPI/blob/3c7b906a63591b293f7b42069c45d0bb30621c9a/index.js#L180
There is also an option to trigger for all tiles on a dashboard (`includeAllTiles: true` for one of more dashboard ids within this list of objects).

Reply:
> Shouldn't this be a list of objects where each object contains a dashboard id and a list of tile ids? i.e. the selected monitors (tiles) can be on different dashboards within the workspace. See https://github.com/squaredup/SquaredUpAPI/blob/3c7b906a63591b293f7b42069c45d0bb30621c9a/index.js#L180 There is also an option to trigger for all tiles on a dashboard (`includeAllTiles: true` for one of more dashboard ids within this list of objects).

Ah, perhaps selected_monitors is the list of objects I'm referring to but still might be missing the shortcut to allow notifications for all tiles of a particular dashboard or dashboards.

Reply:
> I have created the `notify_on` property in which if you use `all_monitors` that should include all tile
> 
> > `notify_on` (String) Condition to trigger the alert. Must be one of: 'workspace_state', 'all_monitors', or 'selected_monitors'

But I believe that's all monitors in the workspace rather than all monitors in one or more dashboards

Reply:
Oh I see, I missed that one and got confused with the other [includeAllTiles](https://github.com/squaredup/SquaredUpAPI/blob/3c7b906a63591b293f7b42069c45d0bb30621c9a/index.js#L207). I will add this.